sydent/users/tokens.py

```diff
@@ -22,6 +23,18 @@
 logger = logging.getLogger(__name__)
 
 def issueToken(sydent, user_id):
+    """
+    Creates an account for the given Matrix user ID, then generates, saves and returns
+    an access token for that account.
+
+    :param sydent: The Sydent instance to use for storing the token.
+    :type sydent: sydent.sydent.Sydent
+    :param user_id: The Matrix user ID to issue a token for.
+    :type user_id: unicode
+
+    :return: The access token for that account.
+    :rtype: str
```

Comment:
hrm. I think we're going to end up tying ourselves in knots if we start using `str` widely, because you'll end up with conditional code that needs to decide whether we're on python 2 or 3 and hence whether you need to `encode`/`decode` it when you need to turn it into a `bytes` or a `unicode`.

So this should probably be be returning a `unicode` ?

Reply:
> So this should probably be be returning a `unicode` ?

Nope, because the function that actually returns the token does `"".join(...)` which returns something of type `str` regardless of the Python version. tbf that could just be fixed by doing `u"".join(...)` instead.

However, annoyingly, that's not the only function that does that, namely all of the database `get[...]` functions return a `str` regardless of the Python version because the SQLite lib returns `str` regardless of the Python version when fetching results from the database :/

Basically, every time I wrote `str` as the return type it's because the function has this issue, and every time I wrote `str` as a param's type it's because the function gets passed the return value of a function with this issue.

Should I fixup the functions that I can fixup, knowing that it'll probably be a small percentage of the ones with this issue?

Reply:
oh god. sorry, you're right; I hadn't realised quite how many `str`s there are floating around.

I do think there might be merit in fixing up utility functions like this to return a `unicode`, but it's certainly hard to see where to stop.

can you merge #241 and i'll have another look at this on its own.

Reply:
Done.